example/ram/compile.lurk

```diff
@@ -0,0 +1,71 @@
+(defmacro or (a b)
+  `(if ,a t ,b))
+
+(define map
+  (lambda (f xs)
+    (if (eq nil xs)
+        nil
+        (cons (f (car xs)) (map f (cdr xs))))))
+
+(define cadr
+  (lambda (xs)
+    (car (cdr xs))))
+
+(define caddr
+  (lambda (xs)
+    (car (cdr (cdr xs)))))
+
+(define cadddr
+  (lambda (xs)
+    (car (cdr (cdr (cdr xs))))))
+
+(define let-bindings cadr)
+(define let-body caddr)
+(define lambda-params cadr)
+(define lambda-body caddr)
+
+(define lookup
+  (lambda (e env success failure)
+    (if (eq nil env)
+        (failure)
+        (if (eq nil (atom (car (car env))))
+            (lookup e (car env) success
+                    (lambda () (lookup e (cdr env) success failure)))
+            (if (eq e (car (car env)))
+                (success (cdr (car env)))
+                (lookup e (cdr env) success failure))))))
+
+(define member
+  (lambda (e xs)
+    (if (eq nil xs)
+        nil
+        (if (eq e (car xs))
+            xs
+            (member e (cdr xs))))))
+
+(define compile-closure
+  (lambda (e params ram env)
+    (let ((init-params (append `(if current-env current-ram eval begin atom car cdr emit quote macroexpand + - / * = eq cons) params))
+          (rec (lambda (e) (compile-closure e init-params ram env)))
+          (rec-params (lambda (e params) (compile-closure e (append init-params params) ram env))))
```

Comment:
Just an efficiency note, if you were to change the argument order of `compile-closure`, the definitions of `rec` and `rec-params` could be simplified by leaning on the partial evaluation.

That is, I think you could write something like
```
(define compile-closure
  (lambda (params ram env e)
    (let ((init-params (append `(if current-env current-ram eval begin atom car cdr emit quote macroexpand + - / * = eq cons) params))
          (rec (compile-closure init-params ram env))
          (rec-params (lambda (params) (compile-closure (append init-params params) ram env))))
…)))
```

I'm not saying you have to so here and now, but I've gotten in the habit of doing that when I can, since it's more concise and will result in better performing code. It's therefore become an idiom which feels aesthetically correct to me. This is an argument for potentially developing the same aesthetic.